apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -110,38 +136,67 @@ const BuildGrid: React.FC<{
   build: Build;
   handleTaskClick: (s: string) => () => void;
   isRightmostBuild: boolean;
-}> = ({ build, handleTaskClick, isRightmostBuild }) => (
-  <BuildContainer
-    onClick={(event: React.MouseEvent) => {
-      handleTaskClick(
-        (event.target as HTMLDivElement)?.getAttribute("status") ?? "",
-      );
-    }}
-  >
-    {build.tasks.map(
-      ({ displayName, displayStatusCache, execution, id, status }) => {
-        // Use status as backup for tasks created before displayStatusCache was introduced
-        const taskStatus = (displayStatusCache || status) as TaskStatus;
-        return (
-          <SquareMemo
-            key={id}
-            data-tooltip={`${displayName} - ${taskStatusToCopy[taskStatus]}`}
-            isRightmostBuild={isRightmostBuild}
-            status={taskStatus}
-            to={getTaskRoute(id, { execution })}
-          />
+}> = ({ build, handleTaskClick, isRightmostBuild }) => {
+  const rowRef = useRef<HTMLDivElement>(null);
+  const { width } = useDimensions<HTMLDivElement>(rowRef);
+
+  const { columnWidth, setColumnWidth } = useBuildVariantContext();
+  useEffect(() => {
+    if (width !== 0 && columnWidth !== width) {
+      setColumnWidth(width);
+    }
+  }, [setColumnWidth, columnWidth, width]);
+
+  return (
+    <BuildContainer
+      ref={rowRef}
+      onClick={(event: React.MouseEvent) => {
+        handleTaskClick(
+          (event.target as HTMLDivElement)?.getAttribute("status") ?? "",
         );
-      },
-    )}
-  </BuildContainer>
-);
+      }}
+    >
+      {build.tasks.map(
+        ({ displayName, displayStatusCache, execution, id, status }) => {
+          // Use status as backup for tasks created before displayStatusCache was introduced
+          const taskStatus = (displayStatusCache || status) as TaskStatus;
+          return (
+            <SquareMemo
+              key={id}
+              data-tooltip={`${displayName} - ${taskStatusToCopy[taskStatus]}`}
+              isRightmostBuild={isRightmostBuild}
+              status={taskStatus}
+              to={getTaskRoute(id, { execution })}
+            />
+          );
+        },
+      )}
+    </BuildContainer>
+  );
+};
+
+const squareWithBorder = SQUARE_SIZE + 2;
+const containerPadding = 16 + 2;
+const containerBorder = 2;
```

Comment:
Super nit but would you mind breaking out these magic vars further? This works nicely but it just scares me... e.g.

```js
const padding = spacing[200];
const border = 1;
const containerPadding = (spacing * 2) + (border * 2);
...
const buildGroupCss = css`
  ...
  border: ${border}px solid ${gray.light2};
  padding: ${padding}px;
`